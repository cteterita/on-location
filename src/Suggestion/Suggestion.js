import React from 'react';
import PropTypes from 'prop-types';

import { stringify } from 'query-string';

import { Link } from 'react-router-dom';

// TODO: Style this
function Suggestion(props) {
  const { displayName, lat, lon } = props;
  // TODO: Make this reload the map if you're already on the results page
  const linkTo = {
    pathname: '/results',
    search: stringify({ lat, lon, zoom: 11 }),
  };
  return (
    <li><Link to={linkTo}>{displayName}</Link></li>
  );
}

Suggestion.propTypes = {
  displayName: PropTypes.string.isRequired,
  lat: PropTypes.number.isRequired,
  lon: PropTypes.number.isRequired,
};

export default Suggestion;
