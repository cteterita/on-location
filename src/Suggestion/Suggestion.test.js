import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

import Suggestion from './Suggestion';

test('renders without errors', () => {
  const displayName = 'Terrapin Crossroads';
  render(<BrowserRouter><Suggestion displayName={displayName} key={1} lat={1} lon={1} /></BrowserRouter>);
  expect(screen.getByRole('listitem')).toMatchSnapshot();
});
