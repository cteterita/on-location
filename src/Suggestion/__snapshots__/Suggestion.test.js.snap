// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without errors 1`] = `
<li>
  <a
    href="/results?lat=1&lon=1&zoom=11"
  >
    Terrapin Crossroads
  </a>
</li>
`;
